pkg/controller/seed-controller-manager/update/update_controller.go

```diff
@@ -116,16 +117,36 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 }
 
 func (r *Reconciler) reconcile(ctx context.Context, cluster *kubermaticv1.Cluster) (*reconcile.Result, error) {
-
+	// If cluster is not healthy yet there is nothing to do.
+	// If it gets healthy we'll get notified by the event. No need to requeue.
 	if !cluster.Status.ExtendedHealth.AllHealthy() {
-		// Cluster not healthy yet. Nothing to do.
-		// If it gets healthy we'll get notified by the event. No need to requeue
 		return nil, nil
 	}
 
-	clusterType := v1.KubernetesClusterType
-	if cluster.IsOpenshift() {
-		clusterType = v1.OpenShiftClusterType
+	clusterType := getClusterType(cluster)
+
+	// Check if cluster has non-empty initial machine deployment request serialized
+	// into its annotations and create it if its there.
+	if request, ok := cluster.Annotations[v1.InitialMachineDeploymentRequestAnnotation]; ok && request != "" {
+		var nodeDeployment *v1.NodeDeployment
+		if err := json.Unmarshal([]byte(request), &nodeDeployment); err != nil {
+			return nil, fmt.Errorf("cannot unmarshal initial machine deployment request for %s cluster: %v", cluster.Name, err)
```

Comment:
Should this be changed in order to avoid throwing an error? For example sending some event and then possibly removing the annotation?

Reply:
For now I think this is sufficient.